iiwa14_no_collision.sdf.in

```diff
@@ -0,0 +1,469 @@
+<?xml version="1.0"?>
```

Comment:
Can you state why you need to vendor in this SDFormat file?

We have this in Drake:
https://github.com/RobotLocomotion/drake/blob/985cf2167fb5c388614137c65796772ad3e98f6d/manipulation/models/iiwa_description/sdf/iiwa14_no_collision.sdf

(albeit with bad package names)

Reply:
The trouble is [the relative paths used for the meshes](https://github.com/RobotLocomotion/drake/blob/985cf2167fb5c388614137c65796772ad3e98f6d/manipulation/models/iiwa_description/sdf/iiwa14_no_collision.sdf#L29). When RViz loads the SDFormat robot description it only has a string, not the path it came from. I hacked around it by templating it to make `file://` URIs.

Reply:
Yeah, understood... but can't that be solved by hacking the package path?

Reply:
>  but can't that be solved by hacking the package path?

By `that` do you mean the vendoring or the relative URLs? Vendoring could be avoided by making an `ament_index` like layout in Drake. Relative URL's remain not useful on the ROS side.

To support `package://some_package_name` URIs in ROS 2 means creating a file `share/ament_index/resource_index/some_package_name`, then putting all the relevant files into `share/some_package_name`. To get Drake's `PackageMap` to recognize it, it also needs a stub `package.xml` in `share/some_package_name/package.xml`. See 68615ac which makes the ur10 demo demo do this.

```
share/
├── ament_index
│   └── resource_index
│       └── packages
│           └── ur10_description
└── ur10_description
    ├── meshes
    │   ├── base.mtl
    │   ├── base.obj
    │   ├── forearm.mtl
    │   ├── forearm.obj
    │   ├── shoulder.mtl
    │   ├── shoulder.obj
    │   ├── upperarm.mtl
    │   ├── upperarm.obj
    │   ├── wrist1.mtl
    │   ├── wrist1.obj
    │   ├── wrist2.mtl
    │   ├── wrist2.obj
    │   ├── wrist3.mtl
    │   └── wrist3.obj
    ├── package.xml
    └── ur10.sdf
```

I'll update the interactive demo to use `package://` URI's on the ROS side too, but the interactive demo is less interesting for showing how to work with Drake's package discovery mechanism because it's using the manipulation station diagram.

Reply:
I forgot to mention this also means setting `AMENT_PREFIX_PATH` to get ROS to see the packages. Launching the UR10 demo now needs:

```
AMENT_PREFIX_PATH="$AMENT_PREFIX_PATH:$(pwd)" python3 ros2_demo.py
AMENT_PREFIX_PATH="$AMENT_PREFIX_PATH:$(pwd)" rviz2 -d view.rviz
```

Reply:
Same treatment to interactive demo in 60e2013

Reply:
Ahh, beautiful! Thank you!

FYI @traversaro Relating https://github.com/RobotLocomotion/drake-external-examples/issues/170